state/StateGetPointerToData.F90

```diff
@@ -0,0 +1,78 @@
+#include "MAPL_ErrLog.h"
+
+module mapl3g_StateGetPointerToData
+
+   use mapl_ErrorHandling
+   use mapl_KeywordEnforcer
+   use esmf
+
+   implicit none
+   private
+
+   public :: StateGetPointerToData
```

Comment:
I forget is this the actual interface name for MAPL2 with `GET_POINTER` being a macro?

I'm not a big fan of the `ToData` part of the name.   Can discuss on Monday.

Reply:
Yes, `include/MAPL_Generic.h` defines the macro `MAPL_GetPointer`
```cpp
#define MAPL_GetPointer ESMFL_StateGetPointerToData
```

I like the `ToData` part - otherwise the first question that comes to mind is, pointer to what? The public interface can skip this part to keep with the current conventions.

Reply:
We'll have to ask @atrayano - I'm unaware of why this would need to be a macro.   It just saves `call ` near as I can tell.  Would prefer not to repeat that - pending advice.

Reply:
Starting to think I don't even like the `Pointer` part of the name, but it is so baked into the legacy of MAPL.   At the very least it will need to be in the backward compatibility layer.

Reply:
#3600